src/scripts/selectors/index.js

```diff
@@ -1,7 +1,18 @@
 /* @flow */
 
 import VideoRecord from 'records/VideoRecords'
+import UserRecord from 'records/UserRecords'
 
 import type { RootState } from 'types/ApplicationTypes'
 
+/* Videos */
 export const getVideo = (state: RootState): ?VideoRecord => state.video
+
+/* Users */
+export const getUser = (state: RootState): ?UserRecord => state.user
+export const isLoggingIn = (state: RootState): ?boolean => (state.user) ? state.user.isLoggingIn : false
+export const isLogged = (state: RootState): ?boolean => {
+  const user = state.user
+  return !!(user && user.email && !isLoggingIn(state))
```

Comment:
`isLogged` is a relatively simple selector, but here is the plan for any selector that is more than just reading a property off of a piece of state (and there was no way for you to know this ahead of time, so no worries):

- We will use the [reselect](https://github.com/reactjs/reselect) library to define these selectors
- This lib exports a function called `createSelector` which takes in other selectors and creates a new selector that takes the outputs of the other selectors as its inputs. And in addition, if its inputs are the same as the last time it was invoked, it will not recompute its return value and return a cached result instead.
- this takes advantage of the immutability of our store, because comparisons of whether or not the previous and current inputs are the same is just a `===` check
- also, complex selectors like this I think should go in separate modules. This one would go in `selectors/UserSelectors.js`.

So, this selector I think should look like this:

```javascript
// selectors/UserSelectors.js
import { createSelector } from 'reselect';

import { getIsLoggingIn } from 'selectors/index';

export const  getIsLoggedIn = createSelector(
  [getUser, getIsLoggingIn], (user, isLoggingIn) => user && user.get('email') && !isLoggingIn
)
```

Let me know if this makes sense!